Missing macos version on OSX

Seeing directories computed like
```
_skbuild/macosx--x86_64-3.9/cmake-install/include/rapids/cub/block/specializations/block_reduce_raking_commutative_only.cuh -> _skbuild/macosx--x86_64-3.9/setuptools/bdist.macosx--x86_64/wheel/
```
i.e. with no proper OSX version: `macosx--x86_64-3.9`

This subsequently causes the final install step to fail:
```
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 281, in get_tag
      plat_name = get_platform(self.bdist_dir)
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 52, in get_platform
      result = calculate_macosx_platform_tag(archive_root, result)
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/macosx_libfile.py", line 399, in calculate_macosx_platform_tag
      base_version = tuple(int(x) for x in base_version.split("."))
    File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.9/site-packages/wheel/macosx_libfile.py", line 399, in <genexpr>
      base_version = tuple(int(x) for x in base_version.split("."))
  ValueError: invalid literal for int() with base 10: ''
```

I am not 100% sure this problem lies in scikit-build specifically. There are so many places that implement or vendor something like `get_platform`. But the first concrete thread I can find to pull on is the skbuild directory paths with the incomplete version, so this seems like the best place to start. 

I have tried using earlier versions of setuptools as well as downgrading skbuild to 0.15 and 0.14. This build used to function on OSX so it's not clear what might have changed where. Please let me know what additional information I can try to collect.

@henryiii yes the assertion is hit after an update to 0.17
```
Building wheels for collected packages: legate.core
  Created temporary directory: /private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-wheel-d25cm0bp
  Destination directory: /private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-wheel-d25cm0bp
  Running command Building wheel for legate.core (pyproject.toml)
  /private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/setuptools/dist.py:547: UserWarning: Normalizing '23.03.00.dev+87.g11ff2f3.dirty' to '23.3.0.dev0+87.g11ff2f3.dirty'
    warnings.warn(tmpl.format(**locals()))
  Traceback (most recent call last):
    File "/Users/bryan/anaconda/envs/py39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/Users/bryan/anaconda/envs/py39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/Users/bryan/anaconda/envs/py39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
      return _build_backend().build_wheel(wheel_directory, config_settings,
    File "/private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 413, in build_wheel
      return self._build_with_temp_dir(['bdist_wheel'], '.whl',
    File "/private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 398, in _build_with_temp_dir
      self.run_setup()
    File "/private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 484, in run_setup
      super(_BuildMetaLegacyBackend,
    File "/private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 335, in run_setup
      exec(code, locals())
    File "<string>", line 23, in <module>
    File "/private/var/folders/g5/0qds8m2n00qcfqkfxggy2jf40000gn/T/pip-build-env-sxw1krg7/overlay/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 571, in setup
      assert version, "CMAKE_OSX_DEPLOYMENT_TARGET must not be empty if specified"
  AssertionError: CMAKE_OSX_DEPLOYMENT_TARGET must not be empty if specified
  error: subprocess-exited-with-error
```

--- a/skbuild/setuptools_wrap.py
+++ b/skbuild/setuptools_wrap.py
@@ -567,8 +567,12 @@ def setup(
         # or the ``cmake_args`` setup option.
         for cmake_arg in cmake_args:
             if "CMAKE_OSX_DEPLOYMENT_TARGET" in cmake_arg:
-                version = cmake_arg.split("=")[1]
-                assert version, "CMAKE_OSX_DEPLOYMENT_TARGET must not be empty if specified"
+                version_arg = cmake_arg.split("=")[1]
+                if version_arg:
+                    version = version_arg
+                else:
+                    msg = "CMAKE_OSX_DEPLOYMENT_TARGET should not be empty if specified, ignoring"
+                    warnings.warn(msg, stacklevel=1)
             if "CMAKE_OSX_ARCHITECTURES" in cmake_arg:
                 machine = cmake_arg.split("=")[1]
                 if set(machine.split(";")) == {"x86_64", "arm64"}:

